### PR TITLE
Feature/fee config as substate

### PR DESCRIPTION
**Pull request by talekhinezh**
## Summary
* Add SystemBoot substate which has a single `CostingParameters` field (will add more in subsequent PRs)
* Removed `system_2`/`CallbackState` from kernel. Replaced with cleaner VM init logic.

### CODE REVIEW

File: radix-engine-tests/tests/kernel/kernel_open_substate.rs [hunk 1]
```diff
@@ -35,21 +35,22 @@ pub fn test_open_substate_of_invisible_package_address() {
...
 
     // Create kernel
     let mut id_allocator = IdAllocator::new(executable.intent_hash().to_hash());
-    let mut system = SystemConfig {
+    let mut system = System {
         blueprint_cache: NonIterMap::new(),
         auth_cache: NonIterMap::new(),
         schema_cache: NonIterMap::new(),
-        callback_obj: Vm {
+        callback: Vm {
```
Comment (iamyulong) [inline radix-engine-tests/tests/kernel/kernel_open_substate.rs:50]: Rename rename `callback` to `vms`?

File: radix-engine/src/kernel/kernel_api.rs [hunk 2]
```diff
@@ -173,9 +173,7 @@ pub trait KernelInvokeApi<C> {
 }
 
 pub struct SystemState<'a, M: KernelCallbackObject> {
-    // TODO: Combine system and system_2
     pub system: &'a mut M,
-    pub system_2: &'a M::CallbackState,
```
Comment (iamyulong) [inline radix-engine/src/kernel/kernel_api.rs:178]: Nice!

File: radix-engine/src/system/system_callback.rs [hunk 3]
```diff
@@ -91,25 +102,64 @@ impl SystemLockData {
...
+    type CallFrameData = Actor;
+    type InitInput = C::InitInput;
+
+    fn init<S: BootStore>(
+        store: &S,
+        executable: &Executable,
+        execution_config: &ExecutionConfig,
+        init_input: C::InitInput,
+    ) -> Result<Self, BootloadingError> {
+        let costing_parameters = {
```
Comment (iamyulong) [inline radix-engine/src/system/system_callback.rs:124]: I assume you will move this to the module init later on?

**Review by iamyulong — APPROVED** [review]
Looks really nice.

File: radix-engine-tests/tests/kernel/kernel_open_substate.rs (hunk 1 above)
Comment (talekhinezh) [reply radix-engine-tests/tests/kernel/kernel_open_substate.rs:50]: The Vm object should still be opaque to the System.

**Pull request merged**